src/main/java/uk/ac/ebi/ega/egafuse/config/EgaFuseApplicationConfig.java

```diff
@@ -65,4 +73,36 @@ public RestTemplate restTemplate(OkHttpClient httpClient) {
         restTemplate.setRequestFactory(new OkHttp3ClientHttpRequestFactory(httpClient));
         return restTemplate;
     }
+
+    @Bean
+    public Token token(@Value("${cred.username}") String username, @Value("${cred.password}") String password,
+            @Value("${ega.userId}") String egaUserId, @Value("${ega.userSecret}") String egaUserSecret,
+            @Value("${ega.userGrant}") String egaUserGrant, @Value("${aai.server.url}") String aaiUrl) {
+        return new Token(new NetHttpTransport(), new JacksonFactory(), username, password, egaUserId, egaUserSecret,
+                egaUserGrant, aaiUrl);
+    }
+
+    @Bean
+    public EgaFileService initEgaFileService(OkHttpClient okHttpClient, @Value("${app.server.url}") String apiURL,
+            Token token) {
+        return new EgaFileService(okHttpClient, apiURL, token);
+    }
+
+    @Bean
+    public EgaDatasetService initEgaDatasetService(OkHttpClient okHttpClient, @Value("${app.server.url}") String apiURL,
+            Token token, EgaFileService egaFileService) {
+        return new EgaDatasetService(okHttpClient, apiURL, token, egaFileService);
+    }
+
+    @Bean
+    public EgaFuse initEgaFuse(@Value("${mountPath}") String mountPath, EgaDatasetService egaDatasetService, EgaFileService egaFileService) {
+        EgaDirectory egaDirectory = new EgaDirectory("");

```

Comment:
Now, this is something I could not see before and now I can see clearly. This logic relies on having a empty directory to do not implement the proper logic on EgaFuse. That is something that needs to be rewritten as it is a hacky solution and will be very hard for a new team member to understand.

Reply:
Done